test/gurka/test_conditional_speedlimit.cc

```diff
@@ -0,0 +1,313 @@
+
+    const std::string ascii_map = R"(
+               M
+              /|\
+             / 3 2
+            /  |  \
+    A------B-------C---1--D
+    E------F-------G------H
+            \  |  /
+             \ | /
+              \|/
+               N
+               |
+               4
+               |
+               O--5--P-6-R
+                         |
+                         7
+                         |
+                         Q
+    )";
+
+    const gurka::ways ways = {
+        {"DCBA",
+         {
+             {"ref", "A20"},
+             {"highway", "motorway"},
+             {"oneway", "yes"},
+             {"maxspeed", "100"},
+             {"maxspeed:conditional", "120 @ (19:00-06:00)"},
+         }},
+        {"EFGH",
+         {
+             {"ref", "A20"},
+             {"highway", "motorway"},
+             {"oneway", "yes"},
+             {"maxspeed", "100"},
+             {"maxspeed:conditional", "120 @ (19:00-06:00)"},
+         }},
+
+        {"MB", {{"highway", "motorway_link"}, {"oneway", "yes"}, {"maxspeed", "100"}}},
+        {"CM", {{"highway", "motorway_link"}, {"oneway", "yes"}, {"maxspeed", "100"}}},
+        {"NG", {{"highway", "motorway_link"}, {"oneway", "yes"}, {"maxspeed", "100"}}},
+        {"FN", {{"highway", "motorway_link"}, {"oneway", "yes"}, {"maxspeed", "100"}}},
+
+        {"MNO",
+         {
+             {"highway", "secondary"},
+             {"maxspeed", "80"},
+             {"maxspeed:conditional", "70 @ (Jun-Aug Sa,Su 08:00-20:00)"},
+         }},
+
+        {"OP", {{"highway", "tertiary"}, {"maxspeed", "50"}}},
+        {"PR",
+         {
+             {"highway", "tertiary"},
+             {"maxspeed", "50"},
+             {"maxspeed:conditional", "30 @ (Mo-Fr 07:00-16:00; Sa 09:00-16:00; Su 09:00-16:00)"},
+         }},
+        {"RQ",
+         {
+             {"highway", "tertiary"},
+             {"maxspeed", "50"},
+             {"maxspeed:conditional", "20 @ (Dec Fr[-1]-Jan Mo[2])"},
+         }},
+    };
+
+    const auto layout = gurka::detail::map_to_coordinates(ascii_map, grid_size_meters);
+    map = gurka::buildtiles(layout, ways, {}, {},
+                            VALHALLA_BUILD_DIR "test/data/conditional_speedlimits",
+                            {{"mjolnir.timezone", {VALHALLA_BUILD_DIR "test/data/tz.sqlite"}}});
+  }
+};
+
+gurka::map ConditionalSpeedlimit::map = {};
+
+/*************************************************************/
+
+TEST_F(ConditionalSpeedlimit, RouteApiProto) {
+  const auto result = gurka::do_action(valhalla::Options::route, map, {"D", "Q"}, "auto",
+                                       {{"/date_time/value", "2024-08-15T21:00"}});
+  ASSERT_EQ(result.trip().routes_size(), 1);
+  ASSERT_EQ(result.trip().routes(0).legs_size(), 1);
+  const auto leg = result.trip().routes(0).legs(0);
+  ASSERT_EQ(leg.node_size(), 8);
+
+  // Regular speed limit are not affected, even if date was passed
+  EXPECT_EQ(leg.node(0).edge().speed_limit(), 100); // DC
```

Comment:
Can you add a call to `gurka::raw::assert_path` (or something like that :smile:) to make sure the path is actually as expected?

Reply:
Added in 12118f1fe45fb9913a3fa2197bcd4c829dcb6b66